File: Java/app/src/main/java/visprolabweek1/Person.java
package main.java.visprolabweek1;

// Konsep: Encapsulation

public class Person {
    private String name;
    private int age;

    // setter untuk mengatur nilai
    public void setName(String name) {
        this.name = name;
    }

    public void setAge(int age){
        if (age < 0) {
            System.out.println("Umur tidak boleh negatif!");
        } else {
            this.age = age;
        }
    }
    
    // Getter untuk baca nilai
    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }
}

File: Dart/dart_application_1/bin/person.dart
// Konsep: Encapsulation
//
// Penjelasan:
// - Data private (dengan _), hanya bisa diakses lewat getter/setter.
// - Bisa tambah validasi.

class Person {
  String _name = '';
  int _age = 0;

  // Setter
  void setName(String name) {
    _name = name;
  }

  void setAge(int age) {
    if (age < 0) {
      print("Umur tidak boleh negatif!");
    } else {
      _age = age;
    }
  }

  // Getter
  String getName() => _name;
  int getAge() => _age;

  // Alternatif: bisa pakai getter/setter sintaks Dart
  // Lihat versi lanjut di bawah (opsional)
}